source/agora/node/Ledger.d

```diff
@@ -230,6 +234,84 @@ public class Ledger
         else
             return null;
     }
+
+    /***************************************************************************
+
+        Write the block on the ledgerFile.
+
+        Params:
+            T = Type of struct to writeFile
+            record = Instance of `T` to serialize
+            dg  = Serialization delegate, when this struct is a nested struct
+
+    ***************************************************************************/
+
+    public void writeBlockFile (T) (auto ref T record) @safe
+    if (is(T == struct))
+    {
+        auto fd = File("ledger.dat", "w"); // TODO move to config.yaml
```

Comment:
Shouldn't this open the file for appending, not writing? Because otherwise you will keep writing one block and overwriting it in this function each time it's called with a block.

Reply:
changed "a" / append access
if the file does not exist, it is created as empty
if the file already exists, its contents are preserved and it is opened to be written at the end